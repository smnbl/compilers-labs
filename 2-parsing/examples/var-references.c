int varrefs()
{
    int var = 4;
    int arr[4];

    int y = var;
    int z = arr[0];
}
