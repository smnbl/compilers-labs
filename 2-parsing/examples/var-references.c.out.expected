└── Program
    └── FuncDecl: returnType = 'int', name = 'varrefs'
        └── CompoundStmt
            ├── VarDecl: type = 'int', name = 'var'
            │   └── IntLiteral: value = '4'
            ├── ArrayDecl: type = 'int', name = 'arr'
            │   └── IntLiteral: value = '4'
            ├── VarDecl: type = 'int', name = 'y'
            │   └── VarRefExpr: name = 'var'
            └── VarDecl: type = 'int', name = 'z'
                └── ArrayRefExpr: name = 'arr'
                    └── IntLiteral: value = '0'
